### PR TITLE
[8.0][account_banking_payment_export] Add a custom view to select journal items that will added on payment order

### PR DESCRIPTION
**Pull request by adrienpeiffer**
- Add a custom journal items view based on an analysis of several production system
- Display credit and debit considering payment mode's type
- Create a new journal entry ref field which will contain the number of the related invoice if it exist and if the state of the releted move is draft. Otherwise, it will contain the name of the related journal entry.

### CODE REVIEW

File: account_banking_payment_export/models/account_move_line.py [hunk 1]
```diff
@@ -19,12 +19,31 @@
 #
 ##############################################################################
 
-from openerp.osv import orm
+from openerp.osv import orm, fields
 
 
 class AccountMoveLine(orm.Model):
```
Comment (pedrobaeza) [inline account_banking_payment_export/models/account_move_line.py]: You can take the occassion to change this class to new API.

File: account_banking_payment_export/wizard/payment_order_create.py [hunk 2]
```diff
@@ -109,6 +109,12 @@ def search_entries(self):
         context = self.env.context.copy()
         context['line_ids'] = self.filter_lines(lines)
         context['populate_results'] = self.populate_results
+        if payment.payment_order_type == 'payment':
```
Comment (pedrobaeza) [inline account_banking_payment_export/wizard/payment_order_create.py:112]: Shorten this with:

```
context['display_credit'] = payment.payment_order_type == 'payment'
context['display_debit'] = payment.payment_order_type != 'payment'
```

Comment (sbidoul) [inline account_banking_payment_export/wizard/payment_order_create.py:112]: I'd say it's a matter of taste. I personally find @adrienpeiffer 's version a tad easier to read.

**Comment by sbidoul** [comment]
This is a huge usability improvement for populating payment orders. Here is a screenshot:

image

:+1: (code review and functional test)

**Comment by adrienpeiffer** [comment]
@sbidoul @pedrobaeza  Thanks for the review ! account_move_line.py is now migrated in new API ! For the comment in payment_order_create.py, I also find it more readable now.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by chrisdec** [comment]
:+1: (functional test)

**Comment by adrienpeiffer** [comment]
Thanks !

**Pull request merged**